src/query/remote/compressed_codecs.go

```diff
@@ -185,6 +190,7 @@ func compressedSeriesFromSeriesIterator(
 	if err != nil {
 		return nil, err
 	}
+	it.Tags().Rewind()
```

Comment:
Better to `defer it.Tags().Rewind()` before call to `buildTags(...)` so even in error case this is reset yeah?

Reply:
Similarly may be worth deferring reader rewinds too, although that will need to be added to a slice of finalizeables since they're inside a loop